dask_expr/tests/test_shuffle.py

```diff
@@ -130,6 +130,39 @@ def test_task_shuffle_index(npartitions, max_branch, pdf):
     assert sorted(df3.compute().values) == list(range(20))
 
 
+def test_shuffle_str_column_not_in_dataframe(df):
+    # ddf = from_pandas(pdf, npartitions=10)
+    with pytest.raises(
+        KeyError,
+        match="Cannot shuffle on",
+    ) as execinfo:
+        df.shuffle(on="z")  # .compute()
+    assert "z" in str(execinfo.value)
+
+
+def test_shuffle_mixed_list_column_not_in_dataframe(df):
+    # not all cols in list are not in dataframe
+    with pytest.raises(
+        KeyError,
+        match="Cannot shuffle on",
+    ) as execinfo:
+        df.shuffle(["x", "z"])
+    assert "z" in str(execinfo.value)
+    assert "x" not in str(execinfo.value)
+
+
+def test_shuffle_list_column_not_in_dataframe(df):
+    # all cols in list are not in dataframe
```

Comment:
We don't need comments here (only for very complicated tests, the test name should be verbose enough to tell us what's going on here)